src/python/m5/ext/pystats/jsonloader.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2020 The Regents of The University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .simstat import SimStat
+from .statistic import Scalar, Distribution, Accumulator
+from .group import Group, Vector
+import json
+
+class JSONLoader(json.JSONDecoder):
+    """
+    Subclass of JSONDecoder that overrides 'object_hook'. Converts JSON object
+    into a SimStat object.
+
+    Usage
+    -----
+    ```
+    from m5.ext.pystats.jsonloader import JSONLoader
+
+    with open(path) as f:
+        simstat_object = json.load(f, cls=JSONLoader)
```

Comment:
I got a little confused here due to the function at https://github.com/darchr/gem5/pull/218/files#diff-7d72dc1dd99fe1127f61d32392d0943663f4187f7249fcfd68c667bdae10530fR74 . Perhaps it's best to mention here that a helper function is provided to use this class in a cleaner way.

Reply:
resolved [here](https://github.com/darchr/gem5/compare/d974742952a4acfd9a5711f35d25cd42e84d0e70..bf98acaa9d8631db3a97f15eb4bd06524d690d68)

Reply:
resolved [here](https://github.com/darchr/gem5/compare/d974742952a4acfd9a5711f35d25cd42e84d0e70..bf98acaa9d8631db3a97f15eb4bd06524d690d68)